activerecord/lib/active_record/connection_adapters.rb

```diff
@@ -31,6 +31,48 @@ def resolve(adapter_name) # :nodoc:
         class_name, path_to_adapter = @adapters[adapter_name.to_s]
 
         unless class_name
+          # To provide better error messages for adapters expecting the pre-7.2 adapter registration API, we attempt
+          # to load the adapter file from the old location which was required by convention, and then raise an error
+          # describing how to upgrade the adapter to the new API.
+          legacy_adapter_path = "active_record/connection_adapters/#{adapter_name}_adapter"
+
+          begin
+            require legacy_adapter_path
+            # If we reach here it means we found the found a file that may be the legacy adapter and should raise.
+            message = if ActiveRecord::ConnectionHandling.instance_methods.include?("#{adapter_name}_connection".to_sym)
+              # If we find the connection method then we care certain it is a legacy adapter.
+              <<~MSG.squish
+                Database configuration specifies '#{adapter_name}' adapter but that adapter has not been registered.
+                Rails 7.2 has changed the way Active Record database adapters are loaded. The adapter needs to be
+                updated to register itself rather than being loaded by convention.
+                Ensure that the adapter in the Gemfile is at the latest version. If it is, then the adapter may need to
+                be updated.
+                See:
+                https://edgeapi.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters.html#method-c-register
```

Comment:
```suggestion
                https://edgeapi.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters.html#method-c-register
```

We probably shouldn't link to `edgeapi`.

Reply:
I'm not sure what you are meaning to suggest, but I guess we can link to just `api` as it will be live once this change is released in gem.

Reply:
Yeah, we only link directly to `edgeapi` when we intentionally want to send someone to the latest version, e.g. the contribute guide. When building the docs, the generator will properly replace `api.` with `edgeapi.` for the main branch.